tools/src/test/java/org/apache/kafka/tools/consumer/ConsoleConsumerTest.java

```diff
@@ -239,4 +284,73 @@ public void shouldWorkWithoutTopicOption() throws IOException {
         verify(mockConsumer).subscribe(any(Pattern.class));
         consumer.cleanup();
     }
+
+    @ClusterTest(brokers = 3)
+    public void testTransactionLogMessageFormatter(ClusterInstance cluster) throws Exception {
+        try (Admin admin = cluster.createAdminClient()) {
+
+            NewTopic newTopic = new NewTopic(topic, 1, (short) 1);
+            admin.createTopics(singleton(newTopic));
+            produceMessages(cluster);
+
+            String[] transactionLogMessageFormatter = new String[]{
+                "--bootstrap-server", cluster.bootstrapServers(),
+                "--topic", Topic.TRANSACTION_STATE_TOPIC_NAME,
+                "--formatter", "org.apache.kafka.tools.consumer.TransactionLogMessageFormatter",
+                "--from-beginning"
+            };
+
+            ConsoleConsumerOptions options = new ConsoleConsumerOptions(transactionLogMessageFormatter);
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            PrintStream output = new PrintStream(out);
+            ConsoleConsumer.process(1, options.formatter(),
```

Comment:
please call `cleanup` to close consumer ... otherwise, @FrankYang0529 will hate your PR due to thread leak

Reply:
Thanks for your comments, I will use try catch block to close these resources